nodebuilder/header/header.go

```diff
@@ -4,32 +4,78 @@ import (
 	"context"
 
 	"github.com/celestiaorg/celestia-node/header"
+	libhead "github.com/celestiaorg/celestia-node/libs/header"
 )
 
 // Module exposes the functionality needed for querying headers from the network.
 // Any method signature changed here needs to also be changed in the API struct.
 //
 //go:generate mockgen -destination=mocks/api.go -package=mocks . Module
 type Module interface {
+	// Head returns the ExtendedHeader of the chain head.
+	Head(context.Context) (*header.ExtendedHeader, error)
+
+	// Get returns the header of the given hash from the node's header store.
+	Get(ctx context.Context, hash libhead.Hash) (*header.ExtendedHeader, error)
+	// GetVerifiedRangeByHeight returns the given range [from:to) of ExtendedHeaders
+	// from the node's header store and verifies that the returned headers are
+	// adjacent to each other.
+	GetVerifiedRangeByHeight(
+		ctx context.Context,
+		from *header.ExtendedHeader,
+		to uint64,
+	) ([]*header.ExtendedHeader, error)
 	// GetByHeight returns the ExtendedHeader at the given height, blocking
 	// until header has been processed by the store or context deadline is exceeded.
 	GetByHeight(context.Context, uint64) (*header.ExtendedHeader, error)
-	// Head returns the ExtendedHeader of the chain head.
-	Head(context.Context) (*header.ExtendedHeader, error)
+
 	// IsSyncing returns the status of sync
 	IsSyncing(context.Context) bool
+	// WaitSync blocks until the header Syncer is synced to network head.
+	WaitSync(ctx context.Context) error
+	// SyncHead provides the Syncer's view of the current network head.
+	SyncHead(ctx context.Context) (*header.ExtendedHeader, error)
+
+	// Subscribe creates long-living Subscription for validated ExtendedHeaders.
+	// Multiple Subscriptions can be created.
```

Comment:
```suggestion
	// Subscribe creates long-living Subscription for newly validated ExtendedHeaders.
	// Multiple Subscriptions can be created.
```

Reply:
removed this endpoint for now as mocks don't support generics yet.

Reply:
tracked in #1474